rq_dashboard/web.py

```diff
@@ -223,13 +228,37 @@ def get_queue_registry_jobs_count(queue_name, registry_name, offset, per_page):
     return (total_items, jobs)
 
 
-@blueprint.route("/")
-@blueprint.route("/view")
-@blueprint.route("/view/queues")
-def queues_overview():
+def escape_format_instance_list(url_list):
+    if isinstance(url_list, (list, tuple)):
+        url_list = [re.sub(r"://:[^@]*@", "://:***@", x) for x in url_list]
+    elif isinstance(url_list, string_types):
+        url_list = [re.sub(r"://:[^@]*@", "://:***@", url_list)]
+    return url_list
+
+
+def change_rq_instance(instance_number):
+    redis_url_list = current_app.config.get("RQ_DASHBOARD_REDIS_URL")
+    redis_url_list = escape_format_instance_list(redis_url_list)
+    if isinstance(redis_url_list, (list, tuple)):
+        _, current_url = from_url(redis_url_list[int(instance_number)])
+        if current_app.redis_conn != current_url:
+            pop_connection()
+            current_app.redis_conn = current_url
+            push_rq_connection()
+    return redis_url_list
```

Comment:
We can avoid this by properly setting up a request and using instance_number provided by request from [view_args](https://flask-doc.readthedocs.io/en/latest/api.html#flask.Request.view_args)